services/painting-surface-defect-detection-model-service/app/routers/predict.py

```
@@ -0,0 +1,198 @@
+        # 이미지 데이터 읽기
+        image_data = await image.read()
+        
+        # 예측 수행
+        result = predict_anomaly(image_data, confidence_threshold)
+        
+        logger.info(f"Prediction completed successfully for {image.filename}")
+        return result
+        
+    except Exception as e:
+        logger.error(f"Error in prediction: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Prediction failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)